stdlib/public/core/VarArgs.swift

```diff
@@ -64,52 +59,15 @@ let _x86_64SSERegisterWords = 2
 let _x86_64RegisterSaveWords = _x86_64CountGPRegisters + _x86_64CountSSERegisters * _x86_64SSERegisterWords
 #endif
 
-/// Invoke `f` with a C `va_list` argument derived from `args`.
-public func withVaList<R>(args: [CVarArgType],
-  @noescape _ f: CVaListPointer -> R) -> R {
-  let builder = VaListBuilder()
-  for a in args {
-    builder.append(a)
-  }
-  return withVaList(builder, f)
-}
-
-/// Invoke `f` with a C `va_list` argument derived from `builder`.
-public func withVaList<R>(builder: VaListBuilder,
-  @noescape _ f: CVaListPointer -> R) -> R {
-  let result = f(builder.va_list())
-  _fixLifetime(builder)
-  return result
-}
-
-/// Returns a `CVaListPointer` built from `args` that's backed by
-/// autoreleased storage.
-///
-/// - Warning: This function is best avoided in favor of
-///   `withVaList`, but occasionally (i.e. in a `class` initializer) you
-///   may find that the language rules don't allow you to use
-/// `withVaList` as intended.
-@warn_unused_result
-public func getVaList(args: [CVarArgType]) -> CVaListPointer {
-  let builder = VaListBuilder()
-  for a in args {
-    builder.append(a)
-  }
-  // FIXME: Use some Swift equivalent of NS_RETURNS_INNER_POINTER if we get one.
-  Builtin.retain(builder)
-  Builtin.autorelease(builder)
-  return builder.va_list()
-}
-
 @warn_unused_result
 public func _encodeBitsAsWords<T : CVarArgType>(x: T) -> [Int] {
   let result = [Int](
     count: (sizeof(T.self) + sizeof(Int.self) - 1) / sizeof(Int.self),
     repeatedValue: 0)
   var tmp = x
-  _memcpy(dest: UnsafeMutablePointer(result._baseAddressIfContiguous),
-          src: UnsafeMutablePointer(Builtin.addressof(&tmp)),
-          size: UInt(sizeof(T.self)))
+  withUnsafeMutablePointer(&tmp){
+    UnsafeMutablePointer(result._baseAddressIfContiguous).moveAssignFrom($0,count:sizeof(T.self)/sizeof(Int.self))
```

Comment:
`moveAssignFrom` would corrupt memory here if `T` has a non-trivial deinitializer.

Also, you are changing code semantics here -- imagine that `sizeof(T)` is not a multiple of `sizeof(Int)`.  We used to not read bytes beyond the point where `T` ends.  In the new code, we are reading those bytes.